metadata/ota_metadata/metadata_gen.py

```diff
@@ -111,6 +117,54 @@ def ignore_rules(target_dir, ignore_file):
     return parser
 
 
+def _get_latest_kernel_version(boot_dir: Path):
+    kfiles_path = str(boot_dir / "vmlinuz*")
+
+    pa = re.compile(r"vmlinuz-(?P<version>\d+\.\d+\.\d+-\d+)(?P<suffix>.*)")
+
+    def compare(l, r):
+        ma_l = pa.match(Path(l).name)
+        ma_r = pa.match(Path(r).name)
+        ver_l = version.parse(ma_l["version"])
+        ver_r = version.parse(ma_r["version"])
+        return 1 if ver_l > ver_r else -1
+
+    kfile_glob = [f for f in glob.glob(kfiles_path) if not Path(f).is_symlink()]
+    kfiles = sorted(kfile_glob, key=cmp_to_key(compare), reverse=True)
+
+    return Path(kfiles[0])  # latest
+
+
+def _list_non_latest_kernels(boot_dir: Path):
+    kfiles_path = str(boot_dir / "vmlinuz*")
+    ifiles_path = str(boot_dir / "initrd.img*")
+    sfiles_path = str(boot_dir / "System.map*")
+    cfiles_path = str(boot_dir / "config*")
```

Comment:
I am a little bit worried about using `config*` as pattern to search file under boot_dir, what about using more precise pattern like `config-*.*.*-*-*`.